docs/src/what_is_rush.md

```diff
@@ -0,0 +1,24 @@
+## 1. What is Rush?
+
+Rush is a Rust implementation of the [Aleph Consensus Protocol](https://arxiv.org/abs/1908.05156) that offers a convenient API allowing it to be easily applied to various problems. The prime application of Rush is to be the consensus engine (sometimes called the "finality gadget") of the [Aleph Zero blockchain](https://alephzero.org/).
+
+### 1.1 High level idea of what Rush does.
+From the high level perspective Rush allows a set of `N` prespecified nodes to agree on an ordering of items arriving in some unreliable streams of data that these nodes locally observe. An illustrative example to have in mind here is when these nodes all observe a growing blockchain that does not have a built-in notion of finality, and would like to finalize blocks. Then the above mentioned "streams of data" are simply the advancing sequences of blockchain "tips" that each of the nodes sees. Note that in this (and in all the interesting examples), because of possible forks, or network delays, the data streams of individual nodes might not be consistent.
+The goal of a consensus protocol, is to ensure consistency of the decisions, even though relying on an unreliable data source. Consequently, what Rush produces is a single stream of data that "combines" all the individual streams of the `N` nodes and importantly **is consistent** among the nodes. Thus, in the example above, all the nodes would produce a unique sequence of **finalized** blocks (see also below for a more detailed description on how to use Rush as a finality gadget for a blockchain).
```

Comment:
(see also below for a more detailed description on how to use Rush as a finality gadget for a blockchain) it's not belowe but in other chapter so maybe it would be good to add a link to it? 
https://stackoverflow.com/questions/51187658/markdown-reference-to-section-from-another-file/51207259

Reply:
I added references in all the places I spotted.